src/worker/prettier-instance-worker.js

```diff
@@ -14,81 +12,84 @@ function requireInstance(modulePath) {
   return prettierInstance;
 }
 
-parentPort.on("message", ({ type, payload }) => {
-  switch (type) {
-    case "import": {
-      const { modulePath } = payload;
-      try {
-        const prettierInstance = requireInstance(modulePath);
-        parentPort.postMessage({
-          type,
-          payload: { version: prettierInstance.version },
-        });
-      } catch {
-        parentPort.postMessage({
-          type,
-          payload: { version: null },
-        });
+function normalizeResult(methodName, id, result) {
+  // For prettier-vscode, `languages` are enough
+  if (methodName === "getSupportInfo") {
+    result = { languages: result.languages };
+  }
+  return {
+    type: "callMethod",
+    payload: { result, id, isError: false },
+  };
+}
+
+module.exports = (parentPort) => {
```

Comment:
This function is basically just a wrapped version of the existing message handlers that were already in the class, with a small DRY'd up `normalizeResult` utility to avoid some existing code duplication.